### PR TITLE
Split the workflow section and remove outdated advices

### PR DESCRIPTION
**Pull request by alalazo**
This PR removes a few outdated sections from the "Basics" part of the documentation. It also makes a few topic under the environment section more prominent by removing an unneeded `spack.yaml` subsection and promoting everything under it.

The ToC under the "Basics" part after the PR looks like:
Screenshot from 2022-03-14 15-57-24

The "Environment" section ToC looks instead:

Screenshot from 2022-03-14 16-02-47

### CODE REVIEW

File: lib/spack/docs/travis_ci.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
+.. Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+   Spack Project Developers. See the top-level COPYRIGHT file for details.
+
+   SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+========================
+Using Spack on Travis-CI
```
Comment (sethrj) [inline lib/spack/docs/travis_ci.rst]: I thought that recent decisions by the Travis corporation meant it's basically useless for open source builds these days, and developers have been shifting to GitHub actions etc. Is this section still relevant?

**Review by sethrj — DISMISSED** [review]
Looks good, just one comment about the travis-ci still being there

File: lib/spack/docs/travis_ci.rst (hunk 1 above)
Comment (alalazo) [reply lib/spack/docs/travis_ci.rst]: I was not sure about that, but if there is consensus I can remove this section too

Comment (haampie) [reply lib/spack/docs/travis_ci.rst]: Yeah, I think travis is effectively a paid service now.

File: lib/spack/docs/replace_conda_homebrew.rst [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+First, let's create a new environment. We'll assume that Spack is already set up
+correctly, and that you've already sourced the setup script for your shell.
+To create a new environment, simply run:
+
+.. code-block:: console
+
+   $ spack env create myenv
+   ==> Updating view at /Users/me/spack/var/spack/environments/myenv/.spack-env/view
+   ==> Created environment 'myenv' in /Users/me/spack/var/spack/environments/myenv
+   $ spack env activate myenv
```
Comment (haampie) [inline lib/spack/docs/replace_conda_homebrew.rst]: One issue with activating and then adding specs is that you don't get the environment variables for the packages you've added & installed.

File: lib/spack/docs/replace_conda_homebrew.rst [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+-----
+
+First, let's create a new environment. We'll assume that Spack is already set up
+correctly, and that you've already sourced the setup script for your shell.
+To create a new environment, simply run:
+
+.. code-block:: console
+
+   $ spack env create myenv
+   ==> Updating view at /Users/me/spack/var/spack/environments/myenv/.spack-env/view
```
Comment (haampie) [inline lib/spack/docs/replace_conda_homebrew.rst]: we should get rid of this info message, doesn't make sense.

File: lib/spack/docs/travis_ci.rst (hunk 1 above)
Comment (alalazo) [reply lib/spack/docs/travis_ci.rst]: Removed in 4b22b4d

**Review by sethrj — DISMISSED** [review]

File: lib/spack/docs/replace_conda_homebrew.rst (hunk 3 above)
Comment (haampie) [reply lib/spack/docs/replace_conda_homebrew.rst]: see #29536

File: lib/spack/docs/environments.rst [hunk 4]
```diff
@@ -833,9 +822,47 @@ regenerate`` will regenerate the views for the environment. This will
...
+there is an entry for ``all`` and no other entry is satisfied by the
+spec. Where the keyword ``all`` appears in the file does not
+matter.
+
+Given the example above, any spec satisfying ``zlib@1.2.8``
+will be linked into ``/my/view/zlib-1.2.8/``, any spec satisfying
+``hdf5@1.8.10+mpi %gcc@4.9.3 ^mvapich2@2.2`` will be linked into
+``/my/view/hdf5-1.8.10/mvapich2-2.2-gcc-4.9.3``, and any spec
+satisfying ``hdf5@1.8.10~mpi %gcc@4.9.3`` will be linked into
+``/my/view/hdf5-1.8.10/gcc-4.9.3``.
```
Comment (haampie) [inline lib/spack/docs/environments.rst]:
```suggestion
Given the example above, the spec ``zlib@1.2.8``
will be linked into ``/my/view/zlib-1.2.8/``, the spec
``hdf5@1.8.10+mpi %gcc@4.9.3 ^mvapich2@2.2`` will be linked into
``/my/view/hdf5-1.8.10/mvapich2-2.2-gcc-4.9.3``, and the spec
``hdf5@1.8.10~mpi %gcc@4.9.3`` will be linked into
``/my/view/hdf5-1.8.10/gcc-4.9.3``.
```

It feels a bit weird to talk about *some* spec satisfying another spec, that satisfies the projection spec.

Comment (alalazo) [reply lib/spack/docs/environments.rst]: Addressed all your comments in 4338aa7 (though I want to note that I moved the content verbatim).

File: lib/spack/docs/travis_ci.rst (hunk 1 above)
Comment (adamjstewart) [reply lib/spack/docs/travis_ci.rst]: Wow I didn't realize that Travis CI is no longer free for open-source projects! RIP Travis CI...

File: lib/spack/docs/replace_conda_homebrew.rst [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+You can see the packages we added earlier in the ``specs:`` section. If you
+ever want to add more packages, you can either use ``spack add`` or manually
+edit this file.
+
+We also need to change the ``concretization:`` option. By default, Spack
+concretizes each spec *separately*, allowing multiple versions of the same
+package to coexist. Since we want a single consistent environment, we want to
+concretize all of the specs *together*.
+
+Here is what your ``spack.yaml`` looks like with this new settings:
```
Comment (adamjstewart) [inline lib/spack/docs/replace_conda_homebrew.rst]:
```suggestion
Here is what your ``spack.yaml`` looks like with this new setting:
```

File: lib/spack/docs/replace_conda_homebrew.rst [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+
+.. code-block:: diff
```
Comment (adamjstewart) [inline lib/spack/docs/replace_conda_homebrew.rst]:
```suggestion
.. code-block:: yaml
```

File: lib/spack/docs/replace_conda_homebrew.rst [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+a list of packages we would like to install into our environment. Let's say we
+want a newer version of Bash than the one that comes with macOS, and we want a
+few Python libraries. We can run:
+
+.. code-block:: console
+
+   $ spack -e myenv add bash python@3 py-numpy py-scipy py-matplotlib
+
+Each package can be listed on a separate line, or combined into a single line like we did above.
+Notice that we're explicitly asking for Python 3 here. You can use any spec
```
Comment (adamjstewart) [inline lib/spack/docs/replace_conda_homebrew.rst]: Not sure if `python@3` is really necessary anymore. When I first wrote this, Spack still defaulted to Python 2. If we want to make it clear that specs are allowed, maybe we could say `bash@5`? That shows that we are telling Spack to install a newer version of bash than the one on the OS. Or we could just drop any version specification.

**Review by adamjstewart — COMMENTED** [review]
I agree with all of the removed sections here. It's crazy that so much of these docs haven't been updated in 6 years. I think we're approaching the point where Spack's documentation is actually _too_ extensive and no one reads it because it would take weeks. Making things more succinct and removing cruft should help with this.

File: lib/spack/docs/replace_conda_homebrew.rst (hunk 7 above)
Comment (alalazo) [reply lib/spack/docs/replace_conda_homebrew.rst]: I'll go with `bash@5`. I kept `python@3` to show a spec with a constraint.

Comment (alalazo) [reply lib/spack/docs/replace_conda_homebrew.rst]: See 94656e3

File: lib/spack/docs/replace_conda_homebrew.rst (hunk 6 above)
Comment (alalazo) [reply lib/spack/docs/replace_conda_homebrew.rst]: See 94656e3

File: lib/spack/docs/replace_conda_homebrew.rst (hunk 5 above)
Comment (alalazo) [reply lib/spack/docs/replace_conda_homebrew.rst]: See 94656e3

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**